apis/v1alpha2/shared_types.go

```diff
@@ -261,81 +141,20 @@ const (
-	// longer necessary.
-	ControllerName GatewayController `json:"controllerName"`
-
-	// Conditions describes the status of the route with respect to the Gateway.
-	// Note that the route's availability is also subject to the Gateway's own
-	// status conditions and listener status.
-	//
-	// If the Route's ParentRef specifies an existing Gateway that supports
-	// Routes of this kind AND that Gateway's controller has sufficient access,
-	// then that Gateway's controller MUST set the "Accepted" condition on the
-	// Route, to indicate whether the route has been accepted or rejected by the
-	// Gateway, and why.
-	//
-	// A Route MUST be considered "Accepted" if at least one of the Route's
-	// rules is implemented by the Gateway.
-	//
-	// There are a number of cases where the "Accepted" condition may not be set
-	// due to lack of controller visibility, that includes when:
-	//
-	// * The Route refers to a non-existent parent.
-	// * The Route is of a type that the controller does not support.
-	// * The Route is in a namespace the controller does not have access to.
-	//
-	// +listType=map
-	// +listMapKey=type
-	// +kubebuilder:validation:MinItems=1
-	// +kubebuilder:validation:MaxItems=8
-	Conditions []metav1.Condition `json:"conditions,omitempty"`
-}
+// +k8s:deepcopy-gen=false
+type RouteParentStatus = v1beta1.RouteParentStatus
 
 // RouteStatus defines the common attributes that all Routes MUST include within
 // their status.
-type RouteStatus struct {
-	// Parents is a list of parent resources (usually Gateways) that are
-	// associated with the route, and the status of the route with respect to
-	// each parent. When this route attaches to a parent, the controller that
-	// manages the parent must add an entry to this list when the controller
-	// first sees the route and should update the entry as appropriate when the
-	// route or gateway is modified.
-	//
-	// Note that parent references that cannot be resolved by an implementation
-	// of this API will not be added to this list. Implementations of this API
-	// can only populate Route status for the Gateways/parent resources they are
-	// responsible for.
-	//
-	// A maximum of 32 Gateways will be represented in this list. An empty list
-	// means the route has not been attached to any Gateway.
-	//
-	// +kubebuilder:validation:MaxItems=32
-	Parents []RouteParentStatus `json:"parents"`
-}
+// +k8s:deepcopy-gen=false
+type RouteStatus = v1beta1.RouteStatus
 
 // Hostname is the fully qualified domain name of a network host. This matches
 // the RFC 1123 definition of a hostname with 2 notable exceptions:
 //
-// 1. IPs are not allowed.
-// 2. A hostname may be prefixed with a wildcard label (`*.`). The wildcard
-//    label must appear by itself as the first label.
+//  1. IPs are not allowed.
+//  2. A hostname may be prefixed with a wildcard label (`*.`). The wildcard
+//     label must appear by itself as the first label.
```

Comment:
this is an unneeded diff.

Reply:
fixed by @shaneutt in https://github.com/kubernetes-sigs/gateway-api/pull/1569.